### PR TITLE
docs: stage changelog for release

### PR DESCRIPTION
**Pull request by DanStough**
### Description
Refresh changelog prior to release. Expecting this to sync to ENT before promoting the release.

### CODE REVIEW

**Review by zalimeni — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ IMPROVEMENTS:
 * dns: new version was not supporting partition or namespace being set to 'default' in CE version. [[GH-21230](https://github.com/hashicorp/consul/issues/21230)]
 * mesh: update supported envoy version 1.29.4 in addition to 1.28.3, 1.27.5, 1.26.8. [[GH-21142](https://github.com/hashicorp/consul/issues/21142)]
 * upgrade go version to v1.22.3. [[GH-21113](https://github.com/hashicorp/consul/issues/21113)]
+* upgrade go version to v1.22.4. [[GH-21265](https://github.com/hashicorp/consul/issues/21265)]
```
Comment (zalimeni) [inline CHANGELOG.md:27]: ~ Nit: we should combine these two
```suggestion
* upgrade go version to v1.22.4. [[GH-21265](https://github.com/hashicorp/consul/issues/21265)]
```

**Review by dhiaayachi — APPROVED** [review]
LGTM!

**Comment by dhiaayachi** [comment]
@DanStough shouldn't we add a changelog for the fix to use text/template vs html/template?

**Comment by DanStough** [comment]
@dhiaayachi good question. Was that a regression introduced in 1.19 or before? If before I say yes. If it was in the release I say no.

**Comment by dhiaayachi** [comment]
yeah good point! the issue is only in 1.19 so no need to I guess 🤷

**Pull request merged**